pkg/kubevirt-api-lifecycle-automation/application.go

```diff
@@ -0,0 +1,246 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The Kubevirt Authors
+ *
+ */
+
+package kubevirt_api_lifecycle_automation
+
+import (
+	"context"
+	"fmt"
+	"github.com/kubevirt/kubevirt-api-lifecycle-automation/pkg/client"
+	envmanager "github.com/kubevirt/kubevirt-api-lifecycle-automation/pkg/kubevirt-api-lifecycle-automation/env-manager"
+	"github.com/kubevirt/kubevirt-api-lifecycle-automation/pkg/patch"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/kubevirt/kubevirt-api-lifecycle-automation/pkg/log"
+	"k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	virtv1 "kubevirt.io/api/core/v1"
+)
+
+const (
+	machineTypeEnvName     = "MACHINE_TYPE_GLOB"
+	namespaceEnvName       = "NAMESPACE"
+	restartRequiredEnvName = "RESTART_REQUIRED"
+	labelSelectorEnvName   = "LABEL_SELECTOR"
+)
+
+type MachineTypeUpdater struct {
+	virtClient      client.KubevirtApiLifecycleAutomationClient
+	machineTypeGlob string
```

Comment:
the field name is a bit confusing.. can i suggest you leave a comment for an example of the value it shell contain
if it's a single value like `pc-q35-rhel10.4.0` or a regex value?

Reply:
Turns out glob pattern is a thing i never knew at the time i wrote this comment, so for now i think it's suffice.
If you want to be extra readable, rename it to `machineTypeGlobPattern`.
But i don't think it's that important now